src/formatDate.js

```diff
@@ -8,7 +8,34 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldSeparator = fromFormat.slice(3);
+  const newSeparator = toFormat.slice(3);
+  const splitDate = date.split(oldSeparator);
+
+  const newDate = [];
+  const objectDate = {};
+
+  for (let i = 0; i < splitDate.length; i++) {
+    objectDate[fromFormat[i]] = splitDate[i];
+  }
+
+  if (fromFormat.includes('YY') && toFormat.includes('YYYY')) {
+    if (objectDate['YY'] < 30) {
+      objectDate['YYYY'] = '20' + objectDate['YY'];
+    } else {
+      objectDate['YYYY'] = '19' + objectDate['YY'];
+    }
+  }
```

Comment:
This block of code can be simplified. You only need to check if you are converting from 'YY' to 'YYYY' once and then assign the correct century based on the condition. There's no need to assign 'YYYY' inside the condition blocks.